x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -115,7 +123,28 @@ static DeprecationIssue checkUniqueRealmOrders(final Settings settings, final Pl
         );
     }
 
-    static DeprecationIssue checkImplicitlyDisabledBasicRealms(final Settings settings, final PluginsAndModules pluginsAndModules) {
+    static DeprecationIssue checkImplicitlyDisabledSecurityOnBasicAndTrial(final Settings settings,
+                                                                           final PluginsAndModules pluginsAndModules,
+                                                                           final XPackLicenseState licenseState) {
+        if ( settings.hasValue(XPackSettings.SECURITY_ENABLED.getKey()) == false
+            && (licenseState.getOperationMode().equals(License.OperationMode.BASIC)
+            || licenseState.getOperationMode().equals(License.OperationMode.TRIAL))) {
+          String details = "The behavior where the value of [xpack.security.enabled] setting is false for " +
+              licenseState.getOperationMode() + " licenses is deprecated and will be changed in a future version." +
+              "See https://www.elastic.co/guide/en/elasticsearch/reference/" + Version.CURRENT.major + "." +
+              Version.CURRENT.minor + "/security-minimal-setup.html to enable security, or explicitly disable security by " +
+              "setting [xpack.security.enabled] to false in elasticsearch.yml";
+            return new DeprecationIssue(
+                DeprecationIssue.Level.WARNING,
```

Comment:
I can be definitely swayed. The way I read `WARNING` / `CRITICAL` is that the "Failures will occur unless" meant to convey that i.e. the node will fail to start or the cluster won't form etc. So I went with WARNING. 

It will probably matter a lot how this is surfaced in the upgrade assistant though and if warning is something users are allowed to click through while upgrading, we should make this CRITICAL. Thank for raising this

Reply:
I suspect we can revisit this during the final 8.0 upgrade preparations and make sure it's presenting well to the user.